lib/libc/stdio/lib_tempnam.c

```diff
@@ -105,7 +105,12 @@ FAR char *tempnam(FAR const char *dir, FAR const char *pfx)
 	FAR char *path;
 	int ret;
 
-	(void)asprintf(&path, "%s/%s-XXXXXX.tmp", dir, pfx);
+	if(pfx == NULL) {
+		(void)asprintf(&path, "%s/XXXXXX.tmp", dir);
```

Comment:
I think if pfx is suppose to point to a string of 5 bytes.   when it's null, i guess it may need to be XXXXX.tmp ?  and when it's not null, we may have to restrict for 5 bytes only.  but it's appended with pfx+6 byte string name. what is your opinion ?

Reply:
Void type casting needed??

Reply:
@pradeepns85 
 and when it's not null, we may have to restrict for 5 bytes only. but it's appended with pfx+6 byte string name:
pfx can be any length but suffix is fixed to 6 bytes.
For null case we have only 6 bytes